jobbergate-cli/jobbergate_cli/subapps/applications/tools.py

```diff
@@ -420,32 +421,129 @@ def load_application_from_source(app_source: str, app_config: JobbergateApplicat
     return application
 
 
-def execute_application(
-    app_module: JobbergateApplicationBase,
-    app_config: JobbergateApplicationConfig,
-    supplied_params: Optional[Dict[str, Any]] = None,
-    fast_mode: bool = False,
-):
+@dataclass
+class ApplicationRuntime:
     """
-    Execute the jobbergate application python module.
-
-    Updates the app_config with values gathered in the question workflow
-
-    :param: app_module:      The source code for the application to execute
-    :param: app_config:      The configuration for the JobbergateApplication
-    :param: supplied_params: Pre-set values for the parameters. Any questions about these values will be skipped.
-    :param: fast_mode:       If true, do not ask the user questions. Just use supplied_params or defaults
-    :returns: The configuration values collected from the user by executing the application
+    Prepare and execute a Jobbergate application gathering the answers to the questions.
+
+    :param app_data: The application data, can be either an ApplicationResponse or a LocalApplication.
+    :type app_data: Union[ApplicationResponse, LocalApplication]
```

Comment:
The type information is already available in the function signature. i don't think the type docs are necessary here.

Reply:
addressed in [8b678e2](https://github.com/omnivector-solutions/jobbergate/pull/489/commits/8b678e2a717bbb82f195e3a158127707909196c4)